File: LC03.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l = 0; r = 0
        d = Counter()
        n = len(s)
        ans = 0

        while r < n:
            d[s[r]] += 1
            
            while d[s[r]] > 1:
                d[s[l]] -= 1
                l += 1
            
            ans = max(ans, r - l + 1)

            r += 1
        
        return ans
            
            

File: LC03.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l = 0, n = s.size(), len = 0;
        unordered_map <char, int> m;
        bool flag;

        for(int r = 0; r < n; r++){
            m[s[r]]++;

            while(m[s[r]] > 1){
                m[s[l++]]--;
            }

            len = max(len, r - l + 1);
        }

        return len;
    }
};
